common/device_parser.cc

```diff
@@ -107,9 +107,9 @@ bool DeviceCache::readFromCache(const std::string& device_id, std::map<std::stri
   return false;
 }
 
-void DeviceCache::addToCache(const std::string& device_id, const std::map<std::string, std::string> device_attrs)
+void DeviceCache::addToCache(const std::string& device_id, std::map<std::string, std::string>&& device_attrs)
 {
   WriteLock wl(&d_rwlock);
 
-  d_devicemap.emplace(std::make_pair(device_id, std::move(device_attrs)));
+  d_devicemap.emplace(device_id, std::move(device_attrs));
```

Comment:
I think the std::move here is redundant since it's already an rvalue?

Reply:
Yes of course!

Reply:
Hum no, `device_attrs` is considered as a lvalue in the current expression thus not moved by default.

Reply:
But map::emplace is defined as:
`template <class... Args>  pair<iterator,bool> emplace (Args&&... args);`

So this shouldn't this "just work" if device_attrs is already a universal reference?

Reply:
I just looked it up and I'm wrong :)

I wonder if we should use std::forward rather than move here? Then the caller is in control

Reply:
In fact I'm positive we should use std::forward here, see my other comment

Reply:
The other option is simply to get rid of all the move stuff, because I'm just adding to a cache, and there's absolutely no reason not to copy it